vm/rust/src/jsonrpc.rs

```diff
@@ -58,6 +65,79 @@ struct StateDiff {
     replaced_classes: Vec<ReplacedClass>,
 }
 
+impl StateDiff {
+    pub fn from_state_maps(state_maps: &StateMaps) -> Self {
+        Self {
+            storage_diffs: state_maps
+                .storage
+                .iter()
+                .fold(
+                    IndexMap::<StarkFelt, Vec<Entry>>::new(),
+                    |mut acc, ((address, key), value)| {
+                        let starkfelt_address = Felt::from(address.clone()).into(); // Convert ContractAddress -> StarkFelt
+                        let entry = Entry {
+                            key: Felt::from(key.clone()).into(), // Convert StorageKey -> StarkFelt
+                            value: Felt::from(value.clone()).into(), // Convert Felt -> StarkFelt
+                        };
+
+                        acc.entry(starkfelt_address)
+                            .or_insert_with(Vec::new)
+                            .push(entry);
+
+                        acc
+                    },
+                )
+                .into_iter()
+                .map(|(address, storage_entries)| StorageDiff {
+                    address,
+                    storage_entries,
+                })
+                .collect(),
+
+            nonces: state_maps
```

Comment:
The same comments as before (in the context of this function) apply to all the following fields declaration